### DISCUSSION TITLE
I have a suggestion.

### DISCUSSION THREAD

**Discussion by AkshatChauhan18** [PowerShell/PowerShell#15306] [General]
I don't whether it is possible or not , powershell should have feature for embedding like xterm have.
I am a python programmer and I know that we can embed xterm in PyQt and tkinter but not powershell.
This an example of embedding xterm - 
`from Tkinter import *
import os

root = Tk()
termf = Frame(root, height=400, width=500)

termf.pack(fill=BOTH, expand=YES)
wid = termf.winfo_id()
os.system('xterm -into %d -geometry 40x20 -sb &' % wid)

root.mainloop()`
I want that powershell should also have feature like this.

**Comment by vexx32** [comment]
Python embedding directly is probably not going to be that straightforward since pwsh is a .NET application, but it might be worth your time to look at Snek, which apparently allows interfacing between Python and pwsh in some fashion.

**Comment by KalleOlaviNiemitalo** [comment]
How about adding `-e pwsh` to the xterm command so that xterm runs pwsh instead of your default shell?

```python
os.system('xterm -into %d -geometry 40x20 -sb -e pwsh &' % wid)
```
